### PR TITLE
Tiny fixes

### PR DESCRIPTION
**Pull request by ryanio**
This PR combines some tiny fixes from #905, including:

* Add blockchain leveldb object support info https://github.com/ethereumjs/ethereumjs-vm/pull/972#issuecomment-736740557
* Add structure diagram edit URL Mermaid chart below the diagram to make the update procedure more transparent
* VM: Add better description for addresses used along VM.runCode(), see #979
* Add some dev documentation along the comment from @ryanio https://github.com/ethereumjs/ethereumjs-monorepo/pull/951#issuecomment-727250776
* Common: Fix README typo from https://github.com/ethereumjs/ethereumjs-monorepo/pull/1034#discussion_r553517646
* Permalink updates ethereumjs-vm -> ethereumjs-monorepo
* Add `Trie`, `Util` to issue templates

This PR additionally:
* Updates typedoc to latest, removes some deprecated options
* Fixes devp2p peer communication examples with new tx library updates

I haven't regenerated the docs yet since it looks like it will change a few hundred files (since the `library` mode was removed)  and I wanted to make the review easier, so I can add that as a last commit after review, or before the next release, whichever you'd prefer @holgerd77.

### CODE REVIEW

**Comment by ryanio** [comment]
client karma run is having some trouble with a downstream update: `Uncaught TypeError: TextDecoder is not a constructor`

it looks like it's from [`web-encoding v1.1.2`](https://github.com/Gozala/web-encoding/releases/tag/v1.1.2) which was just released, i'm looking into if it's an issue on our end or theirs.

**Comment by holgerd77** [comment]
@ryanio Cool! 😄  👍 

Maybe don't regenerate the docs, I'd rather do this along the PR on the releases, so if we merge earlier here than I would do this on the release PR, then we are more flexible here and I would really like to get these releases out this week. If I will do the releases before the merge here it is also better to wait since it otherwise would get a bit tricky on rebase (I've also changed a lot of docs there).

**Comment by ryanio** [comment]
ok this should be ready for review now! feel free to merge if it looks good.

File: packages/vm/typedoc.js [hunk 1]
```diff
@@ -11,7 +10,6 @@ module.exports = {
     'lib/evm/**',
     'lib/state/cache.ts',
   ],
-  excludeNotExported: true,
```
Comment (holgerd77) [inline packages/vm/typedoc.js:14]: Do you have got a clear reason to have this removed? I haven't got the complete oversight of the consequences right now.

File: packages/block/package.json [hunk 2]
```diff
@@ -58,10 +58,10 @@
     "eslint-config-prettier": "^6.11.0",
     "eslint-plugin-implicit-dependencies": "^1.0.4",
     "istanbul": "^0.4.1",
-    "karma": "^5.2.1",
-    "karma-browserify": "^7.0.0",
+    "karma": "^6.3.2",
```
Comment (holgerd77) [inline packages/block/package.json:61]: Out of interest: have you looked if this has any practical consequences (performance, test quality or realism 😄 ,...)?

File: package.json [hunk 3]
```diff
@@ -15,7 +15,7 @@
   "scripts": {
     "postinstall": "npm run bootstrap",
     "bootstrap": "lerna bootstrap --ignore-scripts --include-dependencies --no-ci --hoist && npm run temporaryClientFixes && npm run build",
-    "temporaryClientFixes": "cp node_modules/interface-datastore/dist/src/key.d.ts node_modules/interface-datastore/src && cd node_modules/bcrypto && npm i",
+    "temporaryClientFixes": "rm -r node_modules/acorn && cp -r node_modules/karma-typescript/node_modules/acorn/ node_modules/acorn/ && cp node_modules/interface-datastore/dist/src/key.d.ts node_modules/interface-datastore/src && cd node_modules/bcrypto && npm i",
```
Comment (holgerd77) [inline package.json:18]: Can you give a (very short) description what is happening here (or have I missed?)?

File: packages/client/package.json [hunk 4]
```diff
@@ -102,6 +102,7 @@
     "karma-firefox-launcher": "^2.1.0",
     "karma-tap": "^4.2.0",
     "karma-typescript": "^5.5.1",
+    "karma-typescript-es6-transform": "^5.5.1",
```
Comment (holgerd77) [inline packages/client/package.json:105]: Is this introduction related to the quoted problem?

> client karma run is having some trouble with a downstream update: `Uncaught TypeError: TextDecoder is not a constructor`
> 
> it looks like it's from [`web-encoding v1.1.2`](https://github.com/Gozala/web-encoding/releases/tag/v1.1.2) which was just released, i'm looking into if it's an issue on our end or theirs.

**Review by holgerd77 — APPROVED** [review]
Looks good, thanks, nice clean-up! 😄  👍

File: packages/vm/typedoc.js (hunk 1 above)
Comment (ryanio) [reply packages/vm/typedoc.js:14]: yes this option was removed:

> This is mentioned in the release notes here: https://github.com/TypeStrong/typedoc/releases/tag/v0.20.0
> > Removed options - mode, excludeNotExported, includeDeclarations, ignoreCompilerErrors, entryPoint

Comment (ryanio) [reply packages/vm/typedoc.js:14]: the real impacting change is that the experimental `library` mode was removed and there is a new default mode. so our newly regenerated docs after these changes may look quite different in structure and markup, but should hopefully be an improvement and align with the future development of the package.

File: packages/block/package.json (hunk 2 above)
Comment (ryanio) [reply packages/block/package.json:61]: i was updating the deps in trying to resolve some miscellaneous issues but it was all working fine so i decided to keep the version bumps :)

File: packages/client/package.json (hunk 4 above)
Comment (ryanio) [reply packages/client/package.json:105]: yes, so after I reported my bug they did a new release updating the browser file but it was an esmodules (es6) file so I needed this plugin to transform it to es5 to work in our pipeline.

File: package.json (hunk 3 above)
Comment (ryanio) [reply package.json:18]: yes I should have noted this, thanks.

this is a small temporary workaround fix for an updated `acorn` version which `karma-typescript` needs for the `es6-transform` plugin (which i described here: https://github.com/ethereumjs/ethereumjs-monorepo/pull/1177#discussion_r607922114). karma was hanging on the transform step when using an older version of `acorn`. our node_modules has several `acorn` version but the one on the top level that was being used was outdated, so i copy the newer version located inside `karma-typescript/node_modules` to the top level `node_modules` which resolved the problem.

i'll check back in a few weeks and if the issue is resolved (most likely by other modules updating their `acorn` dependency version) i can remove this temporary workaround 👍
